single-line-diagram/single-line-diagram-core/src/main/java/com/powsybl/sld/layout/LegBusSet.java

```diff
@@ -99,82 +86,36 @@ Set<InternCellSide> getInternCellSides() {
-            extendedNodeSet.addAll(busNodes.stream().filter(Objects::nonNull).collect(Collectors.toList()));
-        } else {
-            LOGGER.error("ExtendedNodeSet inconsistent with NodeBusSet");
-        }
-    }
-
-    Set<BusNode> getExtendedNodeSet() {
-        return extendedNodeSet;
-    }
-
-    static List<LegBusSet> createLegBusSets(VoltageLevelGraph graph, Map<BusNode, Integer> nodeToNb, boolean handleShunts) {
+    static List<LegBusSet> createLegBusSets(VoltageLevelGraph graph, Map<BusNode, Integer> busToNb) {
         List<LegBusSet> legBusSets = new ArrayList<>();
 
         List<ExternCell> externCells = graph.getExternCellStream()
                 .sorted(Comparator.comparing(Cell::getFullId)) // if order is not yet defined & avoid randomness
                 .collect(Collectors.toList());
 
-        if (handleShunts) {
-            manageShunts(graph, externCells, legBusSets, nodeToNb);
-        }
-
-        externCells.forEach(cell -> pushLBS(legBusSets, new LegBusSet(nodeToNb, cell)));
+        externCells.forEach(cell -> pushLBS(legBusSets, new LegBusSet(busToNb, cell)));
 
         graph.getInternCellStream()
                 .filter(cell -> cell.checkIsShape(InternCell.Shape.ONE_LEG))
                 .sorted(Comparator.comparing(Cell::getFullId)) // if order is not yet defined & avoid randomness
-                .forEachOrdered(cell -> pushLBS(legBusSets, new LegBusSet(nodeToNb, cell, Side.UNDEFINED)));
+                .forEachOrdered(cell -> pushLBS(legBusSets, new LegBusSet(busToNb, cell, Side.UNDEFINED)));
 
         graph.getInternCellStream()
                 .filter(cell -> cell.checkIsNotShape(InternCell.Shape.ONE_LEG, InternCell.Shape.UNHANDLED_PATTERN))
                 .sorted(Comparator.comparing(cell -> -((InternCell) cell).getBusNodes().size())         // bigger first to identify encompassed InternCell at the end with the smaller one
                         .thenComparing(cell -> ((InternCell) cell).getFullId()))                        // avoid randomness
-                .forEachOrdered(cell -> pushNonUnilegInternCell(legBusSets, nodeToNb, cell));
+                .forEachOrdered(cell -> pushNonUnilegInternCell(legBusSets, busToNb, cell));
 
         // find orphan busNodes and build their LBS
         List<BusNode> allBusNodes = new ArrayList<>(graph.getNodeBuses());
         allBusNodes.removeAll(legBusSets.stream().
                 flatMap(legBusSet -> legBusSet.getBusNodeSet().stream()).collect(Collectors.toList()));
         allBusNodes.stream()
                 .sorted(Comparator.comparing(Node::getId))              //avoid randomness
-                .forEach(busNode -> legBusSets.add(new LegBusSet(nodeToNb, busNode)));
+                .forEach(busNode -> legBusSets.add(new LegBusSet(busToNb, busNode)));
         return legBusSets;
     }
 
-    private static void manageShunts(VoltageLevelGraph graph, List<ExternCell> externCells, List<LegBusSet> legBusSets, Map<BusNode, Integer> nodeToNb) {
```

Comment:
just tested on another branch (the one dealing with star shunts) with a more complex case, and in fact the result is not the same if removing this or not. It doesn't mean that this method is useful (the result with this manageShunts method is uglier in fact), but it seems that it wasn't doing the same as `externCells.forEach(cell -> pushLBS(legBusSets, new LegBusSet(busToNb, cell)));`. Is this normal?

Reply:
It is unclear to me as well.

Reply:
I really think it should not change the expected behavior. It's ugly. Let's remove it.